dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/DecodeableRpcResult.java

```diff
@@ -125,6 +126,10 @@ public void decode() throws Exception {
 
     private void handleValue(ObjectInput in) throws IOException {
         try {
+            if (invocation instanceof RpcInvocation && serializationType == 2) {
+                setValue(in.readObject(((RpcInvocation) invocation).getReturnType()));
```

Comment:
When the method signature is like `CompletableFuture<String> sayHello(String value);`,  `invocation.getReturnType()` will return `CompletableFuture.class` while `RpcUtils.getReturnType` will return `String.class`